File: killProcess.py
class Solution(object):
    def killProcess(self, pid, ppid, kill):
        """
        :type pid: List[int]
        :type ppid: List[int]
        :type kill: int
        :rtype: List[int]
        """
        n = len(pid)
        adjlst = {}
        
        for i in xrange(n):
            if adjlst.has_key(ppid[i]):
                adjlst[ppid[i]].append(i)
            else:
                adjlst[ppid[i]] = [i]
        result = [kill]
        
        def dfs(x):
            if not adjlst.has_key(x):
                return
            for i in adjlst[x]:
                temp = pid[i]
                result.append(temp)
                dfs(temp)
        
        dfs(kill)
        return result

File: killProcess.java
public class Solution {
    public List<Integer> killProcess(List<Integer> pid, List<Integer> ppid, int kill) {
        int i = 0;
        List<Integer> result = new ArrayList<Integer>();
        result.add(kill);
        
        while (i < ppid.size()) {
            if (ppid.get(i) == kill) {
                if (ppid.contains(pid.get(i))) result.addAll(killProcess(pid, ppid, pid.get(i)));
                else result.add(pid.get(i));
            }
            i++;
        }
        return result;
    }
}